### ISSUE TITLE
Cannot send mail to gmail.com [mail for gmail.com loops back to myself]

### ISSUE THREAD

**Issue by rayures** [tomav/docker-mailserver#1586]
## Context

when sending a email to *@gmail.com it fails. mailer-deamon send a notification to my inbox. All other domains so far do work (hotmail, company email, etc). 

last mail send to *gmail.com was in may (worked). so something changes since then, but i do not know what. I did not make any change to the mailserver.

```
This is the mail system at host example.net.

I'm sorry to have to inform you that your message could not
be delivered to one or more recipients. It's attached below.

For further assistance, please send mail to postmaster.

If you do so, please include this problem report. You can
delete your own text from the attached returned message.

                   The mail system

<adressee@gmail.com>: mail for gmail.com loops back to myself
```

sending emails to *@gmail.com does not work.

### *What* is affected by this bug?
sending emails

### *When* does this occur?
always

## *How* do we replicate the issue?

1. send email to *gmail.com

## Expected behavior (i.e. solution)

sending email to @gmail.com works. 

## Your Environment

* Amount of RAM available:
* Mailserver version used:
* Docker version used:
* Environment settings relevant to the config:
* Any relevant stack traces ("Full trace" preferred):

ISP modem (bride mode) -> PFSENSE firewall -> proxmox host -> debian 10.4 VM, with docker and this mailserver. VM has 16GB ram, 10GB in use. 80GB free disk space.
docker version 19.03.5

log from WORKING emailing to hotmail:

1. 
2. Aug  6 10:15:22 mail postfix/smtpd[1128]: connect from localhost[127.0.0.1]
3. Aug  6 10:15:22 mail postfix/smtpd[1128]: E30AC422B47: client=localhost[127.0.0.1]
4. Aug  6 10:15:22 mail postfix/cleanup[2636]: E30AC422B47: message-id=<85cdb13531755f60fbdf651f8d6526c2@example.net>
5. Aug  6 10:15:22 mail postfix/qmgr[1007]: E30AC422B47: from=<rayures@example.net>, size=7026, nrcpt=1 (queue active)
6. Aug  6 10:15:22 mail postfix/smtpd[1128]: disconnect from localhost[127.0.0.1] ehlo=1 mail=1 rcpt=1 data=1 quit=1 commands=5
7. Aug  6 10:15:22 mail amavis[252]: (00252-02) Passed CLEAN {RelayedOpenRelay}, [IP]:58434 [IP] <rayures@example.net> -> <adressee@hotmail.com>, Queue-ID: AEC12422B5C, Message-ID: <85cdb13531755f60fbdf651f8d6526c2@example.net>, mail_id: 1qKYAPFOheRM, Hits: -, size: 6807, queued_as: E30AC422B47, 106 ms
8. Aug  6 10:15:22 mail postfix/smtp[2632]: AEC12422B5C: to=<adressee@hotmail.com>, relay=127.0.0.1[127.0.0.1]:10024, delay=0.23, delays=0.11/0.02/0/0.1, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as E30AC422B47)
9. Aug  6 10:15:22 mail postfix/qmgr[1007]: AEC12422B5C: removed

log from FAILED emailing to gmail:

1.
2. Aug  6 10:08:58 mail postfix/smtp[1130]: connect to gmail-smtp-in.l.google.com[2a00:1450:4013:c07::1a]:25: Cannot assign requested address
3. Aug  6 10:08:58 mail postfix/postscreen[1131]: CONNECT from [127.0.0.1]:32968 to [127.0.0.1]:25
4. Aug  6 10:08:58 mail postfix/postscreen[1131]: WHITELISTED [127.0.0.1]:32968
5. Aug  6 10:08:58 mail postfix/smtpd[1132]: connect from localhost[127.0.0.1]
6. Aug  6 10:08:58 mail opendmarc[213]: ignoring connection from localhost
7. Aug  6 10:08:58 mail postfix/smtp[1130]: warning: host gmail-smtp-in.l.google.com[0.0.0.0]:25 greeted me with my own hostname example.org
8. Aug  6 10:08:58 mail postfix/smtp[1130]: warning: host gmail-smtp-in.l.google.com[0.0.0.0]:25 replied to HELO/EHLO with my own hostname example.org
9. Aug  6 10:08:58 mail postfix/smtp[1130]: 56CFF422B54: to=<TO@gmail.com>, relay=gmail-smtp-in.l.google.com[0.0.0.0]:25, delay=0.11, delays=0.01/0.01/0.1/0, dsn=5.4.6, status=bounced (mail for gmail.com loops back to myself)

**docker config**
```
 mail:
  image: tvial/docker-mailserver:stable
  hostname: mail
  domainname: example.net
  container_name: mail
  restart: always
  ports: 
   - "25:25" 
   - 143:143
   - 465:465
   - 587:587
   - 993:993
  dns: 1.1.1.1 
  volumes:
   - mail/maildata:/var/mail
   - mail/mailstate:/var/mail-state
   - mail/config:/tmp/docker-mailserver
   - cert:/tmp/ssl:ro
   - /etc/localtime:/etc/localtime:ro
  environment:
   - PUID=1000
   - PGID=999
   - DMS_DEBUG=1
   - ENABLE_CLAMAV=0
   - ONE_DIR=0
   - ENABLE_POP3=0
   - ENABLE_FAIL2BAN=1
   - ENABLE_MANAGESIEVE=
   - OVERRIDE_HOSTNAME=example.net
   - POSTMASTER_ADDRESS=
   - POSTSCREEN_ACTION=ignore #enforce is default
   - REPORT_RECIPIENT=rayures@example.net
   - REPORT_SENDER=mailserver@examplen.et
   - REPORT_INTERVAL=weekly
   - SMTP_ONLY=
   - TLS_LEVEL=intermediate 
   - SPOOF_PROTECTION=
   - ENABLE_SRS=0
   - PERMIT_DOCKER=connected-networks
   - VIRUSMAILS_DELETE_DELAY=
   - ENABLE_POSTFIX_VIRTUAL_TRANSPORT=
   - POSTFIX_DAGENT=
   - POSTFIX_MAILBOX_SIZE_LIMIT=
   - POSTFIX_MESSAGE_SIZE_LIMIT=
   - ENABLE_SPAMASSASSIN=0
   - SA_TAG=2.0
   - SA_TAG2=6.31
   - SA_KILL=6.31
   - SA_SPAM_SUBJECT=***SPAM***
   - ENABLE_FETCHMAIL=0
   - FETCHMAIL_POLL=300
   - ENABLE_LDAP=
   - LDAP_START_TLS=
   - LDAP_SERVER_HOST=
   - LDAP_SEARCH_BASE=
   - LDAP_BIND_DN=
   - LDAP_BIND_PW=
   - LDAP_QUERY_FILTER_USER=
   - LDAP_QUERY_FILTER_GROUP=
   - LDAP_QUERY_FILTER_ALIAS=
   - DOVECOT_TLS=
   - DOVECOT_USER_FILTER=
   - DOVECOT_PASS_FILTER=
   - ENABLE_POSTGREY=0
   - POSTGREY_DELAY=300
   - POSTGREY_MAX_AGE=35
   - POSTGREY_TEXT=Delayed by postgrey
   - ENABLE_SASLAUTHD=0
   - SASLAUTHD_MECHANISMS=
   - SASLAUTHD_MECH_OPTIONS=
   - SASLAUTHD_LDAP_SERVER=
   - SASLAUTHD_LDAP_SSL=
   - SASLAUTHD_LDAP_BIND_DN=
   - SASLAUTHD_LDAP_PASSWORD=
   - SASLAUTHD_LDAP_SEARCH_BASE=
   - SASLAUTHD_LDAP_FILTER=
   - SASLAUTHD_LDAP_START_TLS=
   - SASLAUTHD_LDAP_TLS_CHECK_PEER=
   - SASL_PASSWD=
   - SRS_EXCLUDE_DOMAINS=
   - SRS_SECRET=
   - RELAY_HOST=
   - RELAY_PORT=25
   - RELAY_USER=
   - RELAY_PASSWORD=
   - SSL_TYPE=manual
   - SSL_CERT_PATH=/tmp/ssl/example.net.fullchain
   - SSL_KEY_PATH=/tmp/ssl/example.net.key
  cap_add:
   - NET_ADMIN
   - SYS_PTRACE
```

Things i tried: 

log says: 
_Aug  6 10:08:58 mail postfix/smtp[1130]: connect to gmail-smtp-in.l.google.com[2a00:1450:4013:c07::1a]:25: Cannot assign requested address_
looks like ipv6 issue, forced to use ipv4 in main.cf. did not fix issue.

added fqdn in main.cf. did not fix issue.

restored a backup from Juni, did not fix isuse.

**Comment by casperklein** [comment]
> Aug 6 10:08:58 mail postfix/smtp[1130]: warning: host gmail-smtp-in.l.google.com[0.0.0.0]:25 greeted me with my own hostname example.org

That is the reason, why your mails are looping. For some reason, `gmail-smtp-in.l.google.com` is wrongly resolved to `0.0.0.0` which is equal to `127.0.0.1` / `localhost`.

**Comment by rayures** [comment]
> That is the reason, why your mails are looping. For some reason, `gmail-smtp-in.l.google.com` is wrongly resolved to `0.0.0.0` which is equal to `127.0.0.1` / `localhost`.

tnx! that was the hint i needed to solve it :+1: 

for some reason `dig @1.1.1.1 gmail-smtp-in.l.google.com` ipv4 resolves to `0.0.0.0`. 
changed the docker compose dns line to  `dns: 8.8.8.8` and it works again. 

tnx!
